### DISCUSSION TITLE
Setting/choosing embedded terminal

### DISCUSSION THREAD

**Discussion by sivertism** [neovide/neovide#962] [Q&A] [answered]
Hello! 

I'm using Neovide on Windows, and would like to be able to choose which terminal is opened when I use `:terminal`. If I open Neovide directly (e.g. double click some text file) and then start a terminal inside Neovim (`:terminal`), it opens CMD (which kind of fails as it doesn't understand the UNC path).

On the other hand, if I open Neovide from PowerShell, and then `:terminal`, the terminal I get is PowerShell, and the path works correctly.

Is there any way to choose which terminal is opened? The only information I've found so far is to set $TERM, but I'm not sure how I could do that from inside Neovim.

Thanks!

**Comment by Kethku** [comment, marked as answer]
I think the answer here https://vi.stackexchange.com/questions/14646/how-to-use-zsh-by-default-for-terminal-emulator is what you're looking for.

Note: on windows I set powershell as my shell only when I want to open a terminal via a helper function because some extensions depend on cmd being the default for windows compatibility. https://github.com/Kethku/vim-config/blob/main/fnl/module/bindings.fnl#L62-L67

Note Note: this is written in fennel but is basically equivalent to lua.

**Reply by sivertism** [reply]
That's exactly what I was looking for, thanks!
